content/en/docs/reference/access-authn-authz/certificate-signing-requests.md

```diff
@@ -0,0 +1,207 @@
+    1. Permitted x509 extensions - none
+    1. Permitted key usages - exactly `[]string{"key encipherment", "digital signature", "client auth"}`
+    1. Expiration/cert lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
+    1. CA bit allowed/disallowed - not allowed.
+
+1. `kubernetes.io/kubelet-serving`: signs serving certificates that are honored as a valid kubelet serving certificate
+  by the kube-apiserver, but has no other guarantees.
+  Never auto-approved by kube-controller-manager.
+    1. Trust distribution: signed certificates must be honored by the kube-apiserver as valid to terminate connections to a kubelet.
+       The CA bundle is not distributed by any other means.
+    1. Permitted subjects - organizations are exactly `[]string{"system:nodes"}`, common name starts with `"system:node:"`
+    1. Permitted x509 extensions - only DNS and IP SANs are allowed, at least one DNS or IP SAN must be present
+    1. Permitted key usages - exactly `[]string{"key encipherment", "digital signature", "server auth"}`
+    1. Expiration/cert lifetime - minimum of CSR signer or request.
+    1. CA bit allowed/disallowed - not allowed.
+
+1. `kubernetes.io/legacy-unknown`:  has no guarantees for trust at all. Some distributions may honor these as client
+  certs, but that behavior is not standard kubernetes behavior.
+  Never auto-approved by kube-controller-manager.
+    1. Trust distribution: None.  There is no standard trust or distribution for this signer in a kubernetes cluster.
+    1. Permitted subjects - any
+    1. Permitted x509 extensions - honors SAN extensions and discards other extensions.
+    1. Permitted key usages - any
+    1. Expiration/cert lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
+    1. CA bit allowed/disallowed - not allowed.
+
+Today, failures for all of these are only reported in kube-controller-manager-logs.
+
+Distribution of trust happens out of band for these signers.  Any trust outside of those described above are strictly
+coincidental.  For instance, some distributions may honor `kubernetes.io/legacy-unknown` as client-certificates for the
+kube-apiserver, but this is not a standard.
+None of these usages are related to ServiceAccount token secrets `.data[ca.crt]` in any way.  That CA bundle is only
+guaranteed to verify a connection to the kube-apiserver using the default service (`kubernetes.default.svc`).
+
+## Authorization
+
+To allow creating a CertificateSigningRequest and retrieving any CertificateSigningRequest:
+* create certificates.k8s.io certificatesigningrequests
+* get certificates.k8s.io certificatesigningrequests
+
+To allow approving a CertificateSigningRequest:
+* update certificates.k8s.io certificatesigningrequests/approval
+* approve certificates.k8s.io signers <signerNameDomain>/<signerNamePath> or
+  approve certificates.k8s.io signers <signerNameDomain>/*
```

Comment:
These are hard to follow without knowing which is verb/apiGroup/resource. Should these be in the form of RBAC rules (which is realistically what 99% of people will be wanting to construct anyway)?

Reply:
Should we also include `get/list/watch` in `approval` requirements then too? Technically you'll struggle to approve the CSR unless you have a copy of it (unless patching is done instead..)

Reply:
sure